chisel/src/executor.rs

```diff
@@ -357,6 +384,62 @@ fn format_token(token: Token) -> String {
     }
 }
 
+/// Formats a [pt::EventDefinition] into an inspection message
+///
+/// ### Takes
+///
+/// An borrowed [pt::EventDefinition]
+///
+/// ### Returns
+///
+/// A formatted [pt::EventDefinition] for use in inspection output.
+///
+/// TODO: Verbosity option
+fn format_event_definition(event_definition: &pt::EventDefinition) -> Result<String> {
+    let event_name = event_definition.name.as_ref().expect("Event has a name").to_string();
+    let inputs = event_definition
+        .fields
+        .iter()
+        .map(|param| {
+            let name = param
+                .name
+                .as_ref()
+                .map(ToString::to_string)
+                .unwrap_or_else(|| "<anonymous>".to_string());
+            let kind = Type::from_expression(&param.ty)
+                .and_then(Type::into_builtin)
+                .ok_or_else(|| eyre::eyre!("Invalid type in event {event_name}"))?;
+            Ok(ethabi::EventParam { name, kind, indexed: param.indexed })
+        })
+        .collect::<Result<Vec<_>>>()?;
+    let event = ethabi::Event { name: event_name, inputs, anonymous: event_definition.anonymous };
+
+    Ok(format!(
+        "Type: {}\n├ Name: {}\n└ Signature: {:?}",
+        Paint::red("event"),
+        SolidityHelper::highlight(&format!(
+            "{}({})",
+            &event.name,
+            &event
+                .inputs
+                .iter()
+                .map(|param| format!(
+                    "{}{}{}",
+                    param.kind,
+                    if param.indexed { " indexed" } else { "" },
+                    if param.name.is_empty() {
+                        String::default()
+                    } else {
+                        format!(" {}", &param.name)
+                    },
+                ))
+                .collect::<Vec<_>>()
+                .join(", ")
+        ))
```

Comment:
Oh whoops- looks like I left the comma out in my suggestion diff @Bobface, will need to add that in.
```suggestion
        )),
```

Reply:
After this, lgtm!